src/hotspot/os/aix/os_aix.cpp

```diff
@@ -3023,7 +3023,7 @@ static inline time_t get_mtime(const char* filename) {
 int os::compare_file_modified_times(const char* file1, const char* file2) {
   time_t t1 = get_mtime(file1);
   time_t t2 = get_mtime(file2);
-  return t1 - t2;
+  return checked_cast<int>(t1 - t2);
```

Comment:
It seems like checked_cast<> could fail if the file has a bad timestamp far in the future.
```suggestion
  return primitive_compare(t1, t2);
```

Reply:
This is a good idea to use primitive_compare.